hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -582,9 +583,9 @@ private Object invoke(String nsId, int retryCount, final Method method,
    * @return If the exception comes from an unavailable subcluster.
    */
   public static boolean isUnavailableException(IOException ioe) {
-    if (ioe instanceof ConnectException ||
```

Comment:
I had issues triggering these exceptions in tests in the past. Any idea how we can test this?

Reply:
Now I have no idea.
I copied most of the code from `FederationTestUtils#simulateSlowNameNode` and created `FederationTestUtils#simulateThrowSocketExceptionNameNode`, but it did not work as expected.